agent/job_runner.go

```diff
@@ -62,6 +66,14 @@ func (r JobRunner) Create() (runner *JobRunner, err error) {
 	// the Buildkite Agent API
 	runner.logStreamer = LogStreamer{MaxChunkSizeBytes: r.Job.ChunksMaxSizeBytes, Callback: r.onUploadChunk}.New()
 
+	// Prepare a file to recieve the given job environment
+	if file, err := ioutil.TempFile("", fmt.Sprintf("job-env-%s", runner.Job.ID)) ; err != nil {
+		return runner, err
+	} else {
```

Comment:
This doesn't need the else block, see https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow.

Reply:
Without the else block `file` will be out of scope:

```.go
if file, err := ioutil.TempFile("", fmt.Sprintf("job-env-%s", runner.Job.ID)) ; err != nil {
    return runner, err
}
logger.Debug("[JobRunner] Created env file: %s", file.Name())
runner.envFile = file
```
Will fail to compile with `agent/job_runner.go:73:51: undefined: file`.

I could do:

```.go
file, err := foo()
if err != nil { }
```

But that doesn't match the style elsewhere.

Reply:
> But that doesn't match the style elsewhere.

Yup, that's fair. Match the style of the surrounding code and I'll deal with it holistically to match the golang style at some point.